test/rpc_test.py

```diff
@@ -827,3 +827,19 @@ def test_remote_same_worker(self):
             "worker{}".format(dst_rank), my_rref_function, args=(rref_a, rref_b)
         )
         self.assertEqual(rref_c.to_here(), torch.ones(n, n) + 4)
+
+    @requires_process_group_agent
+    def test_sender_exceptions(self):
+        dist.init_process_group(backend="gloo", init_method=self.init_method, timeout=timedelta(seconds=5))
+        rpc.init_model_parallel(
+            self_name="worker%d" % self.rank,
+            backend=TEST_CONFIG.backend,
+            self_rank=self.rank,
+            init_method=self.init_method,
+        )
+
+        if self.rank == 0:
+            time.sleep(0.5)  # to allow worker 1 to exit without joining
```

Comment:
@pietern, I remember you mentioning that we don't really want to `time.sleep` in tests. Is there a better way to do this? The goal here is to let worker 1 exit before worker 0 tries to send an RPC to it, so that an exception is raised, similar to the example in this PR's description.